community/cypher/interpreted-runtime/src/test/scala/org/neo4j/cypher/internal/runtime/interpreted/commands/expressions/StringFunctionsTest.scala

```diff
@@ -159,4 +158,16 @@ class StringFunctionsTest extends CypherFunSuite {
     reverse("\r\n") should equal(stringValue("\n\r"))
     reverse("\uD801\uDC37") should equal(stringValue("\uD801\uDC37"))
   }
+
+  test("splitTests") {
+    def split(x: Any, y: Any) = SplitFunction(Literal(x), Literal(y))(ExecutionContext.empty, QueryStateHelper.empty)
+
+    split("HELLO", "LL") should equal(stringArray("HE", "O"))
+    split("Separating,by,comma,is,a,common,use,case", ",") should equal(stringArray("Separating", "by", "comma", "is", "a", "common", "use", "case"))
+    split("hello", "X") should equal(stringArray("hello"))
+    split("hello", null) should equal(expectedNull)
+    split(null, "hello") should equal(expectedNull)
+    split(null, null) should equal(expectedNull)
+    intercept[CypherTypeException](split(1024, 10))
+  }
```

Comment:
As these TextValue methods are now moved to the values module, we also have to move/add tests for this functionality to the values module.  If we don't we could leave these methods untested when we remove the interpreted runtime for example.

Reply:
Added more tests to `TextValueSpecification` plus added a new test `TextValueTest` that more or less test the same thing as this test.